components/ingest-service/backend/elastic/elastic.go

```diff
@@ -42,22 +42,38 @@ func (es *Backend) Initializing() bool {
 	return !es.initialized
 }
 
-func (es *Backend) DeleteProjectTag(ctx context.Context, projectTagToBeDeleted string) ([]string, error) {
-	log.Debugf("starting project delete project %q", projectTagToBeDeleted)
+func (es *Backend) PurgeProject(ctx context.Context, projectTagToBeDeleted string) error {
+	log.Infof("starting project delete for id %q", projectTagToBeDeleted)
 
 	esNodeJobID, err := es.DeleteNodeProjectTag(ctx, projectTagToBeDeleted)
 	if err != nil {
-		return []string{}, errors.Wrap(err, "Failed to start Elasticsearch Node project tags delete")
+		return errors.Wrap(err, "Failed to start Elasticsearch Node project tags delete")
 	}
 
 	esActionJobID, err := es.DeleteActionProjectTag(ctx, projectTagToBeDeleted)
 	if err != nil {
-		return []string{}, errors.Wrap(err, "Failed to start Elasticsearch Action project tags delete")
+		return errors.Wrap(err, "Failed to start Elasticsearch Action project tags delete")
 	}
 
-	log.Debugf("Started Project delete with node job ID: %q action job ID %q", esNodeJobID, esActionJobID)
+	log.Debugf("Started Project delete with [node job ID: %q, action job ID: %q]", esNodeJobID, esActionJobID)
 
-	return []string{esNodeJobID, esActionJobID}, nil
+	for {
+		time.Sleep(time.Second * 1)
+
+		nodeJobStatus, err := es.JobStatus(ctx, esNodeJobID)
+		if err != nil {
+			return err
+		}
+		actionJobStatus, err := es.JobStatus(ctx, esActionJobID)
+		if err != nil {
+			return err
+		}
+
+		if nodeJobStatus.Completed && actionJobStatus.Completed {
+			break
+		}
+	}
```

Comment:
The ingest service has real code, as the sub-task to process it has already been merged to master, to wit the calls to `DeleteNodeProjectTag` and `DeleteActionProjectTag`.
Unlike teams and tokens, which are done synchronously, these two sub-tasks are done asynchronously, so it needs a busy-wait loop to monitor the progress and only return when both are done.
Wondering, though, if this needs a timeout check?

Reply:
I wonder if there's a good way to notify the user that things are in an odd state if _one_ (or more) _but not all_ tasks complete ... 🤔

Reply:
timeout check seems worthwhile